docs/developers_guide/micro_frontends_in_open_edx.rst

```diff
@@ -16,35 +16,168 @@ To run an MFE locally you must install `node 12 <https://nodejs.org>`_ (with npm
 
-Example ``.env.development``:
+Production build example::
 
-.. code-block:: bash
+  SITE_NAME="My Site Name" npm run build
+
+The above would only set the SITE_NAME - suitable values for the environment variables should be supplied as well.
+
+Required Environment Variables
+==============================
+
+All environment variables in this section are required for Open edX micro-frontends.
+
+A note on micro-frontend-specific environment variables
+-------------------------------------------------------
+
+Individual micro-frontends may depend on other environment variables not on this list.  Those variables will be documented in the README files in individual micro-frontend repositories, but can be specified on the command line in the same way as described above.
+
+Fundamentals
+------------
+
+``NODE_ENV``
+  Example: ``production`` | ``development`` | ``test``
+
+  Used by both the build process and application to determine what type of environment we're running. Valid values are "development", "test", and "production". All production-like environments should use "production", as this removes development warnings and checks, and can enable code optimizations for many dependencies like React.
+
+``PUBLIC_PATH``
+  Example: ``/``
+  Used by both the build process and application to set the path of the application.  Should include an initial forward slash ``/``.  Will default to ``/`` if not specified.
+
+``BASE_URL``
+  Example: ``https://new-mfe.example.com``
+
+  The fully-qualified URL to the micro-frontend being built. Used by frontend-platform as part of authentication, as well as by many applications for internal links.
+
+Internationalization
+--------------------
+
+``LANGUAGE_PREFERENCE_COOKIE_NAME``
+  Example: ``example-language-preference``
+
+  The name of the cookie where the user's language preferences should be stored.
+
+Service URLs
+------------
+
+``CREDENTIALS_BASE_URL``
+  Example: ``https://credentials.example.com``
+
+  The fully-qualified URL to the credentials service in the target environment.
+
+``DISCOVERY_API_BASE_URL``
+  Example: ``https://discovery.example.com``
+
+  The fully-qualified URL of the `Course Discovery Service <https://github.com/edx/course-discovery>`_ for the target environment.
+
+``PUBLISHER_BASE_URL``
+  Example: ``https://publisher.example.com``
+
+  The fully-qualified URL of the `Publisher Micro-frontend <https://github.com/edx/frontend-app-publisher>`_.  Today this is only used by micro-frontends that are not offically supported in any Open edX release, such as the `Support Tools Micro-frontend <https://github.com/edx/frontend-app-support-tools>`_.  **As such, this should be considered deprecated and will be removed from a future release until such time as publisher itself is included.**
```

Comment:
I would err on the side of putting a deprecation warning right at the top -- even like
```
``PUBLISHER_BASE_URL``
Example: ``https://publisher.example.com``
Status: DEPRECATED
```